pkg/container/images/keychain.go

```diff
@@ -0,0 +1,79 @@
+package images
+
+import (
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+// envKeychain implements a keychain that reads credentials from environment variables
+type envKeychain struct{}
+
+// Resolve implements the authn.Keychain interface
+func (*envKeychain) Resolve(target authn.Resource) (authn.Authenticator, error) {
+	registry := target.RegistryStr()
+
+	// Try registry-specific environment variables first
+	// Format: REGISTRY_<NORMALIZED_REGISTRY_NAME>_USERNAME/PASSWORD
+	normalizedRegistry := strings.ToUpper(strings.ReplaceAll(registry, ".", "_"))
+	normalizedRegistry = strings.ReplaceAll(normalizedRegistry, "-", "_")
+
+	username := os.Getenv("REGISTRY_" + normalizedRegistry + "_USERNAME")
+	password := os.Getenv("REGISTRY_" + normalizedRegistry + "_PASSWORD")
+
+	// If registry-specific vars not found, try generic ones
+	if username == "" || password == "" {
+		username = os.Getenv("DOCKER_USERNAME")
+		password = os.Getenv("DOCKER_PASSWORD")
+	}
+
+	// If still not found, try REGISTRY_USERNAME/PASSWORD
+	if username == "" || password == "" {
```

Comment:
Is `REGISTRY_USERNAME` a standard way of setting credentials? I wonder if we have too many options for doing the same thing, given we also have `DOCKER_USERNAME` and `REGISTRY_REGNAME_USERNAME`